radicle/src/cob/thread.rs

```diff
@@ -0,0 +1,484 @@
+use std::collections::btree_map::Entry;
+use std::collections::BTreeMap;
+use std::ops::{ControlFlow, Deref, DerefMut};
+use std::str::FromStr;
+
+use once_cell::sync::Lazy;
+use radicle_crdt as crdt;
+use serde::{Deserialize, Serialize};
+
+use crate::cob::common::Reaction;
+use crate::cob::store;
+use crate::cob::{History, TypeName};
+use crate::crypto::{PublicKey, Signer};
+
+use crdt::clock::LClock;
+use crdt::lwwreg::LWWReg;
+use crdt::lwwset::LWWSet;
+use crdt::redactable::Redactable;
+use crdt::{Change, ChangeId, Semilattice};
+
+/// Type name of a thread.
+pub static TYPENAME: Lazy<TypeName> =
+    Lazy::new(|| FromStr::from_str("xyz.radicle.thread").expect("type name is valid"));
+
+/// Identifies a tag.
+pub type TagId = String;
+/// Alias for `Author`.
+pub type ActorId = PublicKey;
+
+/// A comment on a discussion thread.
+#[derive(Default, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Comment {
+    /// The comment body.
+    body: String,
+    /// Thread or comment this is a reply to.
+    reply_to: Option<ChangeId>,
+}
+
+impl Comment {
+    /// Create a new comment.
+    pub fn new(body: String, reply_to: Option<ChangeId>) -> Self {
+        Self { body, reply_to }
+    }
+}
+
+impl PartialOrd for Comment {
+    fn partial_cmp(&self, _other: &Self) -> Option<std::cmp::Ordering> {
+        None
```

Comment:
Couldn't they in theory be at least equal? :sweat_smile:

Reply:
Hehe true.

Reply:
https://github.com/radicle-dev/heartwood/pull/123/commits/a79c95283b5d3794e57c90f88b8c6a014853a378